x-pack/test/observability_functional/apps/observability/alerts/rules_page.ts

```diff
@@ -0,0 +1,179 @@
+          params: {
+            search: '',
+            numTimes: 5,
+            timerangeUnit: 'm',
+            timerangeCount: 15,
+            shouldCheckStatus: true,
+            shouldCheckAvailability: true,
+            availability: { range: 30, rangeUnit: 'd', threshold: '99' },
+          },
+          consumer: 'alerts',
+          schedule: { interval: '1m' },
+          tags: [],
+          name: 'uptime',
+          rule_type_id: 'xpack.uptime.alerts.monitorStatus',
+          notify_when: 'onActionGroupChange',
+          actions: [],
+        };
+        const logThresholdRule = {
+          params: {
+            timeSize: 5,
+            timeUnit: 'm',
+            count: { value: 75, comparator: 'more than' },
+            criteria: [{ field: 'log.level', comparator: 'equals', value: 'error' }],
+          },
+          consumer: 'alerts',
+          schedule: { interval: '1m' },
+          tags: [],
+          name: 'error-log',
+          rule_type_id: 'logs.alert.document.count',
+          notify_when: 'onActionGroupChange',
+          actions: [],
+        };
+        uptimeRuleId = await createRule(uptimeRule);
+        logThresholdRuleId = await createRule(logThresholdRule);
+        await observability.alerts.common.navigateToRulesPage();
+      });
+      after(async () => {
+        await deleteRuleById(uptimeRuleId);
+        await deleteRuleById(logThresholdRuleId);
+      });
+
+      it('shows the rules table ', async () => {
+        await testSubjects.existOrFail('rulesList');
+        const tableRows = await find.allByCssSelector('.euiTableRow');
+        const rows = await getRulesList(tableRows);
+        expect(rows.length).to.be(2);
+        expect(rows[0].name).to.be('error-log');
+        expect(rows[0].enabled).to.be('Enabled');
+        expect(rows[1].name).to.be('uptime');
+        expect(rows[1].enabled).to.be('Enabled');
+      });
+
+      it('changes the rule status to "disabled"', async () => {
+        await testSubjects.existOrFail('rulesList');
+        await observability.alerts.rulesPage.clickRuleStatusDropDownMenu();
+        await observability.alerts.rulesPage.clickDisableFromDropDownMenu();
+        await retry.waitFor('The rule to be disabled', async () => {
+          const tableRows = await find.allByCssSelector('.euiTableRow');
+          const rows = await getRulesList(tableRows);
+          expect(rows[0].enabled).to.be('Disabled');
+          return true;
+        });
+      });
+    });
+
+    describe('User permissions', () => {
+      it('shows the Create Rule button when user has permissions', async () => {
+        await observability.alerts.common.navigateToRulesPage();
+        await retry.waitFor(
+          'Create rule button',
+          async () => await testSubjects.exists('createRuleButton')
+        );
+      });
+
+      it(`shows the no permission prompt when the user has no permissions`, async () => {
+        await observability.users.setTestUserRole({
```

Comment:
@fkanout Here's the helper I mentioned ealier:

```
await observability.users.setTestUserRole(
          observability.users.defineBasicObservabilityRole({
            logs: ['read'],
          })
        );
```

Btw here's the PR I made use of it https://github.com/elastic/kibana/pull/117264/files#diff-32316eeb66686d880035c437179197c339f952f5fc10adc10d0229548769c32bR131

Sidenote: I noticed that I was using this code to restore the default test role. 

```
 after(async () => {
        await observability.users.restoreDefaultTestUserRole();
      });
```